lib/kernel/src/global.erl

```diff
@@ -2760,19 +2760,12 @@ inform_connection_loss(_Node, #state{}) ->
 %% Volatile send (does not bring up connections and does not
 %% preserve signal order) of Msg to global name server at Node...
 %%
+
 gns_volatile_send(Node, Msg) ->
-    To = {global_name_server, Node},
-    case erlang:send(To, Msg, [nosuspend, noconnect]) of
-        ok ->
-            ok;
-        noconnect ->
-            ok;
-        nosuspend ->
-            _ = spawn(fun () ->
-                              _ = erlang:send(To, Msg, [noconnect])
-                      end),
-            ok
-    end.
+    OldAsyncDist = process_flag(async_dist, true),
+    _ = erlang:send({global_name_server, Node}, Msg, [noconnect]),
```

Comment:
Is there a chance that send/3 throws an exception (and `async_dist` flag does not get flipped back)?

Reply:
If it were to throw an exception, `global` would terminate and bring down the whole node with it.